target_postgres/postgres.py

```diff
@@ -459,7 +463,7 @@ def persist_csv_rows(self,
 
         update_sql = self._get_update_sql(remote_schema['name'],
                                           temp_table_name,
-                                          remote_schema['key_properties'],
+                                          [key.lower() for key in remote_schema['key_properties']],
```

Comment:
I think you also have an issue with other values being canonicalized to `_` by the `canonicalize_identifier` code. I think for your `persist_csv` code you'll need to do the same thing as what you did on line 142.